queue-api/src/main/java/com/example/queueapi/application/service/QueueService.java

```diff
@@ -0,0 +1,54 @@
+package com.example.queueapi.application.service;
+
+import com.example.queueapi.application.service.dto.QueueServiceDto;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.redis.core.StringRedisTemplate;
+import org.springframework.stereotype.Service;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class QueueService {
+    private final String USER_QUEUE_WAIT_KEY = "user:queue:%s:wait";
+    private final String USER_QUEUE_TOKEN = "user-queue-%s-%d";
+
+    private final StringRedisTemplate stringRedisTemplate;
+
+    public QueueServiceDto registerUser(String queue, Long userId) {
+        String token = createdToken(queue, userId);
+        long unixTimestamp = Instant.now().getEpochSecond();
+        Boolean add = stringRedisTemplate.opsForZSet().add(USER_QUEUE_WAIT_KEY.formatted(queue), token, unixTimestamp);
+
+        if (!add) {
+            throw new IllegalArgumentException("");
+        }
+
+        Long rank = stringRedisTemplate.opsForZSet().rank(USER_QUEUE_WAIT_KEY.formatted(queue), token);
+        return new QueueServiceDto(token, rank >= 0 ? rank + 1 : rank);
+    }
+
+    private String createdToken(String queue, Long userId) {
```

Comment:
```suggestion
    private String createToken(String queue, Long userId) {
```

별거 아니지만 함수명으로는 과거형보다는 `createToken` 이 더 어울릴것 같네요 ㅎㅎ